src/SprykerSdk/Acp/Schema/SchemaExtender.php

```diff
@@ -0,0 +1,165 @@
+        $result = new CreateDefaultEndpointsResponseTransfer();
+
+        $process = new Process([
+            $this->acpRootPath . '/vendor/bin/syncapi',
+            'schema:openapi:update',
+            '--openapi-file',
+            $createDefaultEndpointsRequestTransfer->getSchemaFile(),
+            '--openapi-doc-file',
+            $this->getDefaultEndpointsSchemaPath($createDefaultEndpointsRequestTransfer),
+        ]);
+
+        $process->run();
+
+        if (!$process->isSuccessful()) {
+            $result->addError($this->createMessage(sprintf('Updating openapi schema failed, error output: %s', $process->getErrorOutput())));
+
+            return $result;
+        }
+
+        if ($createDefaultEndpointsRequestTransfer->getAddLocal()) {
+            $registryFileCopied = $this->copyRegistryLocalFile($createDefaultEndpointsRequestTransfer);
+            if (!$registryFileCopied) {
+                $result->addError($this->createMessage('Copying local registry file failed'));
+
+                return $result;
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param \Transfer\CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer
+     *
+     * @return string
+     */
+    protected function getDefaultEndpointsSchemaPath(CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer): string
+    {
+        if ($createDefaultEndpointsRequestTransfer->getAddLocal()) {
+            return $this->acpRootPath . '/config/app/api/openapi/registry_reference_local.yml';
+        }
+
+        return $this->acpRootPath . '/config/app/api/openapi/registry_reference_remote.yml';
+    }
+
+    /**
+     * @param \Transfer\CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer
+     *
+     * @return bool
+     */
+    protected function copyRegistryLocalFile(CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer): bool
+    {
+        $registryFilePath = dirname($createDefaultEndpointsRequestTransfer->getSchemaFileOrFail()) . DIRECTORY_SEPARATOR . 'registry.yml';
+
+        return copy($this->acpRootPath . '/config/app/api/openapi/registry.yml', $registryFilePath);
+    }
+
+    /**
+     * @param \Transfer\CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer
+     *
+     * @return \Transfer\MessageTransfer|null
+     */
+    protected function convertConfigurationToSchema(CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer): ?MessageTransfer
+    {
+        $schema = $this->schemaConverter->convertConfigurationToSchema($createDefaultEndpointsRequestTransfer->getConfigurationFileOrFail());
+
+        $process = new Process([
+            $this->acpRootPath . '/vendor/bin/syncapi',
+            'schema:openapi:update',
+            '--openapi-file',
+            $createDefaultEndpointsRequestTransfer->getSchemaFile(),
+            $this->schemaWriter->writeSchemaToJsonString($schema),
```

Comment:
What was the final decision regarding converting to/from json? Did we postpone it?

Reply:
In this specific case we are do not have anything to convert to json - this is configuration response schema and we start with having a OpenApi object and not a file. We can't pass an object to external command and creating a temp file looks like an overhead - that's why it is converted to JSON.